extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitterConfig.java

```diff
@@ -47,19 +65,72 @@
+      @Nullable @JsonProperty("metric.topic") String metricTopic,
+      @Nullable @JsonProperty("alert.topic") String alertTopic,
       @Nullable @JsonProperty("request.topic") String requestTopic,
+      @Nullable @JsonProperty("segment.metadata.topic") String segmentMetadataTopic,
+      @Nullable @JsonProperty("segment.metadata.topic.format") String segmentMetadataTopicFormat,
       @JsonProperty("clusterName") String clusterName,
       @JsonProperty("producer.config") @Nullable Map<String, String> kafkaProducerConfig
   )
   {
     this.bootstrapServers = Preconditions.checkNotNull(bootstrapServers, "bootstrap.servers can not be null");
-    this.metricTopic = Preconditions.checkNotNull(metricTopic, "metric.topic can not be null");
-    this.alertTopic = Preconditions.checkNotNull(alertTopic, "alert.topic can not be null");
-    this.requestTopic = requestTopic;
+    validateEventTypes(eventTypes);
+    validateSegmentMetadataTopicFormat(segmentMetadataTopicFormat);
+    // Unless explicitly overridden, kafka emitter will always emit metrics and alerts
+    this.eventTypes = eventTypes == null ? new HashSet<>(Arrays.asList(METRICS, ALERTS)) : eventTypes;
+
+    // If the topics are configured, but entry is missing in event.type, then add that event to eventTypes
+    checkForMissingEventType(metricTopic, METRICS);
+    checkForMissingEventType(alertTopic, ALERTS);
+    checkForMissingEventType(requestTopic, REQUESTS);
+    checkForMissingEventType(segmentMetadataTopic, SEGMENT_METADATA);
+
+    this.metricTopic = this.eventTypes.contains(METRICS) ? Preconditions.checkNotNull(metricTopic, "metric.topic can not be null") : null;
+    this.alertTopic = this.eventTypes.contains(ALERTS) ? Preconditions.checkNotNull(alertTopic, "alert.topic can not be null") : null;
+    this.requestTopic = this.eventTypes.contains(REQUESTS) ? Preconditions.checkNotNull(requestTopic, "request.topic can not be null") : null;
+    this.segmentMetadataTopic = this.eventTypes.contains(SEGMENT_METADATA) ? Preconditions.checkNotNull(segmentMetadataTopic, "segment.metadata.topic can not be null") : null;
     this.clusterName = clusterName;
     this.kafkaProducerConfig = kafkaProducerConfig == null ? ImmutableMap.of() : kafkaProducerConfig;
+
+    // If segment.metadata.topic.format is not configured, use json as default like other event types
+    this.segmentMetadataTopicFormat = segmentMetadataTopicFormat == null ? JSON : segmentMetadataTopicFormat;
+    // Metadata in non-JSON format must go their own topic
+    if (this.segmentMetadataTopic != null && !JSON.equals(this.segmentMetadataTopicFormat)) {
+      this.segmentMetadataTopic = this.segmentMetadataTopic + "-" + this.segmentMetadataTopicFormat;
+    }
+  }
+
+  private void validateEventTypes(Set<String> eventTypes)
+  {
+    if (eventTypes != null) {
+      List<String> validEvents = Arrays.asList(METRICS, REQUESTS, ALERTS, SEGMENT_METADATA);
```

Comment:
same here, rather than creating a list of types, it might be simpler to make this an enum and let jackson deserialize it.